products/jbrowse-web/src/rootModel.ts

```diff
@@ -249,13 +249,12 @@ export default function RootModel(
           throw new Error(`unknown internet account type ${configuration.type}`)
         }
 
-        const internetAccount = internetAccountType.stateModel.create({
+        const length = self.internetAccounts.push({
           ...initialSnapshot,
           type: configuration.type,
           configuration,
         })
-        self.internetAccounts.push(internetAccount)
-        return internetAccount
+        return self.internetAccounts[length - 1]
```

Comment:
Also, this is so that the internetAccount is already part of the state tree when it's created. That way volatiles that are initialized in the model can access the tree (in Apollo I'm wrapping a volatile OAuthInternetAccount in the ApolloInternetAccount).